app/javascript/components/community/video-grid/useVideoGrid.tsx

```diff
@@ -14,15 +21,36 @@ export type APIResponse = {
   }
 }
 
+export type HandleTrackChangeType = (track: VideoTrack) => void
+
+export type UseVideoGridReturnType = {
+  handleTrackChange: HandleTrackChangeType
+  selectedTrack: VideoTrack
+  resolvedData: APIResponse | undefined
+  latestData: APIResponse | undefined
+  isFetching: boolean
+  order: string
+  setOrder: (order: string) => void
+  page: number
+  setPage: (page: number) => void
+  criteria: string
+  setCriteria: (criteria: string) => void
+  request: ListState
+}
+
 export function useVideoGrid(
   videoRequest: Request,
-  tracks: AutomationTrack[],
-  selectedTrackSlug: string
-): any {
+  tracks: VideoTrack[]
+): UseVideoGridReturnType {
   const initialTrack =
-    tracks.find((track) => track.slug == selectedTrackSlug) || tracks[0]
+    tracks.find(
+      (t) =>
+        t.slug ===
+        new URLSearchParams(window.location.search).get('video_track_slug')
+    ) || tracks[0]
```

Comment:
Could't this use the `videoRequest` object?

Reply:
I think it could, I change this.